lib/AST/NameLookup.cpp

```diff
@@ -2363,6 +2400,222 @@ swift::getDirectlyInheritedNominalTypeDecls(
   return result;
 }
 
+template <typename T>
+static TinyPtrVector<T *> lookupOperatorImpl(
```

Comment:
I wonder if it would be cleaner to factor operator and precedencegroup lookup into its own translation unit. What do you think?

Reply:
Hmm, could do – in fact it could be argued that it belongs in `ModuleNameLookup.cpp`, what do you think?

Reply:
The lookup logic has been specialized enough that I think it's become a service in its own right.

Reply:
You totally don't have to do this now.

Reply:
@CodaFi I've gone ahead and split it into it's own file, how does it look?